test_replication_all fails

https://github.com/dragonflydb/dragonfly/actions/runs/11907164782/job/33180402391#step:6:1013

Assert fail:

```
    async def check():
        await check_all_replicas_finished(c_replicas, c_master)
        hashes = await asyncio.gather(*(SeederV2.capture(c) for c in [c_master] + c_replicas))
>       assert len(set(hashes)) == 1
E       assert 2 == 1
```

--- a/src/server/journal/executor.cc
+++ b/src/server/journal/executor.cc
@@ -83,6 +83,11 @@ void JournalExecutor::SelectDb(DbIndex dbid) {
     auto cmd = BuildFromParts("SELECT", dbid);
     Execute(cmd);
     ensured_dbs_[dbid] = true;
+
+    // TODO: This is a temporary fix for #4146.
+    // For some reason without this the replication breaks in regtests.
+    auto cb = [](EngineShard* shard) { return OpStatus::OK; };
+    shard_set->RunBriefInParallel(std::move(cb));
   } else {
     conn_context_.conn_state.db_index = dbid;
   }

